File: backend/book-library-backend/src/main/java/com/mycompany/book/library/backend/Book.java
package com.mycompany.book.library.backend;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Book {

    private long id;
    private String title;
    private int pageCount;
    @JsonProperty("ISBN")
    private String isbn;
    private boolean checkedOut;

    public Book() {}

    public Book(long id, String title, int pageCount, String isbn, boolean checkedOut) {
        this.id = id;
        this.title = title;
        this.pageCount = pageCount;
        this.isbn = isbn;
        this.checkedOut = checkedOut;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getPageCount() {
        return pageCount;
    }

    public void setPageCount(int pageCount) {
        this.pageCount = pageCount;
    }

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public boolean isCheckedOut() {
        return checkedOut;
    }

    public void setCheckedOut(boolean checkedOut) {
        this.checkedOut = checkedOut;
    }

    @Override
    public String toString() {
        return "Book{" + "title=" + title + ", pageCount=" + pageCount + ", isbn=" + isbn + ", checkedOut=" + checkedOut + '}';
    }

    @Override
    public int hashCode() {
        int hash = 7;
        hash = 59 * hash + (int) (this.id ^ (this.id >>> 32));
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Book other = (Book) obj;
        if (this.id != other.id) {
            return false;
        }
        return true;
    }

}

File: frontend/src/app/book.ts
export class Book {
    private id : number;
    private title : string;
    private pageCount : number;
    private ISBN : string;
    private checkedOut : boolean;

    constructor(id: number, title : string, pageCount : number, ISBN : string, checkedOut? : boolean) {
        this.id = id;
        this.title = title;
        this.pageCount = pageCount;
        this.ISBN = ISBN;
        this.checkedOut = checkedOut || false;
    }

    public getId() : number {
        return this.id;
    }

    public getTitle() : string {
        return this.title;
    }

    public setTitle(title : string) : void {
        this.title = title;
    }

    public getPageCount() : number {
        return this.pageCount;
    }

    public setPageCount(pageCount : number) : void {
        this.pageCount = pageCount;
    }

    public getISBN() : string {
        return this.ISBN;
    }

    public setISBN(ISBN : string) : void {
        this.ISBN = ISBN;
    }

    public isCheckedOut() : boolean {
        return this.checkedOut;
    }

    public setCheckedOut(checkedOut : boolean) : void {
        this.checkedOut = checkedOut;
    }
}
